server/src/main/java/io/crate/expression/scalar/ExtractFunctions.java

```diff
@@ -123,6 +123,37 @@ public static void register(ScalarFunctionModule module) {
                     new UnaryScalar<Number, Period>(signature, boundSignature, DataTypes.INTERVAL, function::apply)
             );
         }
+        // extract(epoch from ...) is different as is returns a `double precision`
+        module.register(
+            Signature.scalar(
+                functionNameFrom(EPOCH),
+                DataTypes.INTERVAL.getTypeSignature(),
+                parseTypeSignature("double precision")
```

Comment:
```suggestion
                DataTypes.DOUBLE.getTypeSignature()
```

Either is fine, but this would make it typed.